src/store/wallet/wallet-selectors.ts

```diff
@@ -12,29 +13,51 @@ import { emptyToken, TokenInterface } from '../../token/interfaces/token.interfa
 import { walletAccountStateToWalletAccount } from '../../utils/wallet-account-state.utils';
 import { WalletRootState, WalletState } from './wallet-state';
 
-export const useHdAccountsListSelector = () =>
+export const useAccountsListSelector = () =>
   useSelector<WalletRootState, WalletAccountInterface[]>(({ wallet }) =>
-    wallet.hdAccounts.map(walletAccountStateToWalletAccount)
+    wallet.accounts.map(walletAccountStateToWalletAccount)
   );
 
+export const useHdAccountListSelector = () =>
+  useSelector<WalletRootState, WalletAccountInterface[]>(({ wallet }) => {
+    const accounts = [];
+    for (const account of wallet.accounts) {
+      if (account.type === AccountTypeEnum.HD_ACCOUNT) {
+        accounts.push(walletAccountStateToWalletAccount(account));
+      }
+    }
+
+    return accounts;
+  });
+
+export const useImportedAccountListSelector = () =>
+  useSelector<WalletRootState, WalletAccountInterface[]>(({ wallet }) => {
+    const accounts = [];
+    for (const account of wallet.accounts) {
+      if (account.type === AccountTypeEnum.IMPORTED_ACCOUNT) {
+        accounts.push(walletAccountStateToWalletAccount(account));
+      }
+    }
+
+    return accounts;
+  });
```

Comment:
I don't think the accounts list is going to be huge.  
Just use `filter` array method instead of `for of`